pkg/kubelet/kubelet_node_status.go

```diff
@@ -505,8 +505,9 @@ func (kl *Kubelet) setNodeAddress(node *v1.Node) error {
 
 		// 1) Use nodeIP if set
 		// 2) If the user has specified an IP to HostnameOverride, use it
-		// 3) Lookup the IP from node name by DNS and use the first non-loopback ipv4 address
-		// 4) Try to get the IP from the network interface used as default gateway
+		// 3) Lookup the IP from node name by DNS and use the first valid IPv4 address
```

Comment:
Is the intent to go through all of the IP addresses and grab the first IPv4 address encountered AND IF NO IPv4 addresses, then take the first valid IPv6 address? If so, then I think the logic below will not do that, but will instead take the first valid IP address. If the intent is to just take the first valid address, then maybe the logic on L524-530 could be simplified?

Reply:
The intent is to grab the first valid (i.e. non-loopback) IPv4 address. If no valid IPv4 addresses exist, grab the first valid (i.e. non-loopback) IPv6 address.